cw-orch/src/interchain/interchain_channel.rs

```diff
@@ -0,0 +1,394 @@
+        &self,
+        from: String,
+    ) -> Result<(Option<CosmTxResponse>, Option<CosmTxResponse>), DaemonError> {
+        let current_channel_creation_a = self
+            .get_channel_creation_ack(from.clone())
+            .await?
+            .get(0)
+            .cloned();
+
+        let current_channel_creation_b = self
+            .get_channel_creation_confirm(from)
+            .await?
+            .get(0)
+            .cloned();
+
+        Ok((current_channel_creation_a, current_channel_creation_b))
+    }
+
+    // We get the last transactions that is related to creating a channel from chain from to the counterparty chain defined in the structure
+    pub async fn get_last_channel_creation_hash(
+        &self,
+        from: String,
+    ) -> Result<(Option<String>, Option<String>), DaemonError> {
+        let (current_channel_creation_a, current_channel_creation_b) =
+            self.get_last_channel_creation(from).await?;
+        Ok((
+            current_channel_creation_a.map(|tx| tx.txhash),
+            current_channel_creation_b.map(|tx| tx.txhash),
+        ))
+    }
+
+    pub async fn find_new_channel_creation_tx(
+        &self,
+        from: String,
+        last_chain_creation_hashes: &(Option<String>, Option<String>),
+    ) -> Result<(CosmTxResponse, CosmTxResponse), DaemonError> {
+        for _ in 0..5 {
+            match self.get_last_channel_creation(from.clone()).await {
+                Ok(tx) => {
+                    if let Some(ack_tx) = tx.0 {
+                        if let Some(confirm_tx) = tx.1 {
+                            if ack_tx.txhash
+                                != last_chain_creation_hashes
+                                    .0
+                                    .clone()
+                                    .unwrap_or("".to_string())
+                                && confirm_tx.txhash
+                                    != last_chain_creation_hashes
+                                        .1
+                                        .clone()
+                                        .unwrap_or("".to_string())
+                            {
+                                return Ok((ack_tx, confirm_tx));
+                            }
+                        }
+                    }
+                    log::debug!("No new TX by events found");
+                    log::debug!("Waiting 10s");
+                    sleep(Duration::from_secs(10)).await;
+                }
+                Err(e) => {
+                    log::debug!("{:?}", e);
+                    break;
+                }
+            }
+        }
+
+        Err(DaemonError::AnyError(anyhow::Error::msg(format!(
+	        "No new channel creation TX newer than (from_tx_hash: {:?}) or (to_tx_hash: {:?}) found",
+	        last_chain_creation_hashes.0, last_chain_creation_hashes.1
+	    ))))
+    }
+
+    pub async fn follow_packet(
+        &self,
+        from: String,
```

Comment:
What is this from? Can you use a type alias to make it clearer?

type BlaBla = String

Reply:
Sure